crates/swc/src/config/mod.rs

```diff
@@ -314,18 +314,21 @@ impl Options {
 
         let mut program = parse(syntax, es_version, is_module)?;
 
+        let mut transform = transform.unwrap_or_default();
+
         // Do a resolver pass before everything.
         //
         // We do this before creating custom passses, so custom passses can use the
         // variable management system based on the syntax contexts.
         if syntax.typescript() {
+            assumptions.set_public_class_fields = !transform.use_define_for_class_fields;
+
+            assumptions.set_class_methods = true;
```

Comment:
I'm not quite sure if this line should be added. It would generate output more like tsc, but it's different from current behaviour

Reply:
I think it's better to mimic tsc for typescript users.